kernel/src/process.rs

```diff
@@ -536,8 +536,10 @@ struct ProcessDebug {
     /// How many syscalls have occurred since the process started.
     syscall_count: usize,
 
-    /// What was the most recent syscall.
-    last_syscall: Option<Syscall>,
+    /// An ordered list of the last five syscalls the process called along with
+    /// the return values the kernel passed back to the app. Index 0 is the most
+    /// recent syscall.
+    last_syscalls: [(Option<Syscall>, Option<isize>); 5],
```

Comment:
This is going to unconditionally add 35 words per application, correct? That seems like a high cost for non-optional debugging functionality.

Does the return value need to be an `Option<isize>` rather than just an `isize`? I don't think it will be terribly confusing to print that `yield` returned `0`.

Reply:
I was more concerned about the case where the kernel panic-ed because of the syscall, and didn't want to report that a return value was set when it actually wasn't.

Reply:
Maybe it should be 3 instead of 5? It reduces the size a little bit and 3 should be enough to figure out what is going on.

Reply:
Configuring this in `kernel/src/config.rs` would leave the choice for everyone's own trade-offs. A size of zero would even be fine by default, as I only expect to use it when debugging specific things. But when debugging, 3 or 5 may not even be enough, and I'd like to be able to easily increase it to 20.